### DISCUSSION TITLE
Providing a ".Abstractions" library

### DISCUSSION THREAD

**Discussion by johnf-jade** [microsoft/RulesEngine#215] [Ideas]
Has there been any consideration to providing a ".Abstractions" library, with models, an IRulesEngine etc but without all the dependencies?

**Comment by abbasc52** [comment]
Hi @johnf-jade , can you share some more information on how it will benefit the community?

**Reply by johnf-jade** [reply]
In the same way that many of the Microsoft.Extensions libraries have a matching .Abstractions so that consumers can incorporate calls to the IRulesEngine into their business logic layer without taking a dependency on Microsoft.RulesEngine and all it's dependencies, e.g. 430KB of FluentValidation. Of course everyone can just make their own abstractions if they want them.
